3_visualize/src/graph_functions.R

```diff
@@ -0,0 +1,70 @@
+  axis(2, at = seq(0,10), las = 1, tck = -0.01)
+  
+  # slight horizontal offsets so the markers don't overlap:
+  offsets <- data.frame(pgdl = c(0.15, 0.5, 3, 7, 20, 30)) %>%
+    mutate(dl = -pgdl, pb = 0, n_prof = n_profs)
+  
+  #This could be turned into a lapply function
+  for (mod in c('pb','dl','pgdl')){
+    mod_data <- filter(eval_data, model_type == mod)
+    mod_profiles <- unique(mod_data$n_prof)
+    for (mod_profile in mod_profiles){
+      d <- filter(mod_data, n_prof == mod_profile) %>% summarize(y0 = min(rmse), y1 = max(rmse), col = unique(col))
+      x_pos <- offsets %>% filter(n_prof == mod_profile) %>% pull(!!mod) + mod_profile
+      lines(c(x_pos, x_pos), c(d$y0, d$y1), col = d$col, lwd = 2.5)
+    }
+    d <- group_by(mod_data, n_prof) %>% summarize(y = mean(rmse), col = unique(col), pch = unique(pch)) %>%
+      rename(x = n_prof) %>% arrange(x)
+    
+    lines(d$x + tail(offsets[[mod]], nrow(d)), d$y, col = d$col[1], lty = 'dashed')
+    points(d$x + tail(offsets[[mod]], nrow(d)), d$y, pch = d$pch[1], col = d$col[1], bg = 'white', lwd = 2.5, cex = 1.5)
+    
+  }
+  
+  points(2.2, 0.79, col = '#7570b3', pch = 23, bg = 'white', lwd = 2.5, cex = 1.5)
+  text(2.3, 0.80, 'Process-Guided Deep Learning', pos = 4, cex = 1.1)
+  
+  points(2.2, 0.94, col = '#d95f02', pch = 22, bg = 'white', lwd = 2.5, cex = 1.5)
+  text(2.3, 0.95, 'Deep Learning', pos = 4, cex = 1.1)
+  
+  points(2.2, 1.09, col = '#1b9e77', pch = 21, bg = 'white', lwd = 2.5, cex = 1.5)
+  text(2.3, 1.1, 'Process-Based', pos = 4, cex = 1.1)
+  
+  dev.off()
+  
+  readr::write_csv(eval_data, file = file.path(outPathdir, 'model_summary_results.csv'))
+  
+  
+  render_data <- list(pgdl_980mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_980mean = filter(eval_data, model_type == 'dl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_980mean = filter(eval_data, model_type == 'pb', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_500mean = filter(eval_data, model_type == 'dl', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_500mean = filter(eval_data, model_type == 'pb', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_100mean = filter(eval_data, model_type == 'dl', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_100mean = filter(eval_data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pgdl_2mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_2mean = filter(eval_data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
```

Comment:
This is a relatively large chunk of very similar code. You could write a helper function here instead.

Reply:
I could... lol. This is where I was saying "screw it" and get through the exercise for time's sake. I didn't feel like simplifying code was part of the exercise so much as restructuring it towards using the targets package.